### DISCUSSION TITLE
Is it possible to call a object method inside a command?

### DISCUSSION THREAD

**Discussion by dscheinin** [pallets/click#2454] [Q&A]
Hi everyone, I'm trying to invoke an object method inside a command but every time I get an error:
This is the code:
```
import click

from commands.CommandA import CommandA
from commands.CommandB import CommandB

projects_option = click.option("--projects", "-p",
                               help="A comma separated project list.")
repos_option = click.option("--repos", "-r",
                            help="A comma separated repos name to be analyzed.")

@click.group()
def cli():
    """
    Repos tools.
    """
    pass

@cli.command("command-a")
@projects_option
@repos_option
def command_a(projects, repos):
    """Do command A stuff."""
    app = CommandA()
    app.do_command_a_stuff(projects, repos)


@cli.command("command-b")
@projects_option
def command_b(projects):
    """Do command B stuff."""
    app = CommandB()
    app.do_command_b_stuff(projects)

if __name__ == '__main__':
     exit(cli())

```
The problem with this code, is that every time I get this error:

`TypeError: 'bool' object is not callable`

Do you know what I'm doing wrong?

I really appreciate any help you can provide.

**Comment by davidism** [comment]
Yes it's possible. The error indicates a problem with your code, not click.

**Reply by dscheinin** [reply]
> Yes it's possible. The error indicates a problem with your code, not click.

Thank you for your reply @davidism 
I checked my classes code and I find there is something weird with the Logger. If I run the classes without using Click, the methods work fine.
In `__init__` I commented the Logger instantiation and then I ran the script with Click and it's run perfectly

Thank you!
